tests/test_raster/test_interpolate.py

```diff
@@ -0,0 +1,568 @@
+from __future__ import annotations
+
+import re
+from typing import Literal
+
+import numpy as np
+import pytest
+import rasterio as rio
+from scipy.interpolate import interpn
+from scipy.ndimage import binary_dilation
+
+import geoutils as gu
+from geoutils import examples
+from geoutils.projtools import reproject_to_latlon
+from geoutils.raster.interpolate import (
+    _dist_nodata_spread,
+    _interp_points,
+    _interpn_interpolator,
+    method_to_order,
+)
+
+
+class TestInterpolate:
+
+    landsat_b4_path = examples.get_path("everest_landsat_b4")
+    aster_dem_path = examples.get_path("exploradores_aster_dem")
+    landsat_b4_crop_path = examples.get_path("everest_landsat_b4_cropped")
+    landsat_rgb_path = examples.get_path("everest_landsat_rgb")
+
+    def test_dist_nodata_spread(self) -> None:
+        """Test distance of nodata spreading computation based on interpolation order."""
+
+        assert _dist_nodata_spread(0, "half_order_up") == 0
+        assert _dist_nodata_spread(0, "half_order_down") == 0
+        assert _dist_nodata_spread(0, 5) == 5
+
+        assert _dist_nodata_spread(1, "half_order_up") == 1
+        assert _dist_nodata_spread(1, "half_order_down") == 0
+        assert _dist_nodata_spread(5, 5) == 5
```

Comment:
I assume you meant (1, 5) instead of (5, 5). Same for the following block (3).

Reply:
Good catch, fixed!